tests/units/anta_tests/test_snmp.py

```diff
@@ -152,4 +152,78 @@
             "messages": ["SNMP contact is not configured."],
         },
     },
+    {
+        "name": "success",
+        "test": VerifySNMPErrors,
+        "eos_data": [
+            {
+                "counters": {
+                    "inVersionErrs": 0,
+                    "inBadCommunityNames": 0,
+                    "inBadCommunityUses": 0,
+                    "inParseErrs": 0,
+                    "outTooBigErrs": 0,
+                    "outNoSuchNameErrs": 0,
+                    "outBadValueErrs": 0,
+                    "outGeneralErrs": 0,
+                },
+            }
+        ],
+        "inputs": {},
+        "expected": {"result": "success"},
+    },
+    {
+        "name": "success-specific-counters",
+        "test": VerifySNMPErrors,
+        "eos_data": [
+            {
+                "counters": {
+                    "inVersionErrs": 0,
+                    "inBadCommunityNames": 0,
+                    "inBadCommunityUses": 0,
+                    "inParseErrs": 0,
+                    "outTooBigErrs": 5,
+                    "outNoSuchNameErrs": 0,
+                    "outBadValueErrs": 10,
+                    "outGeneralErrs": 1,
+                },
+            }
+        ],
+        "inputs": {"error_counters": ["inVersionErrs", "inParseErrs"]},
+        "expected": {"result": "success"},
+    },
+    {
+        "name": "failure-counters-not-found",
+        "test": VerifySNMPErrors,
+        "eos_data": [
+            {
+                "counters": {},
+            }
+        ],
+        "inputs": {},
+        "expected": {"result": "failure", "messages": ["SNMP counter details not found."]},
+    },
+    {
+        "name": "failure-incorrect-counters",
+        "test": VerifySNMPErrors,
+        "eos_data": [
+            {
+                "counters": {
+                    "inVersionErrs": 1,
+                    "inBadCommunityNames": 0,
+                    "inBadCommunityUses": 0,
+                    "inParseErrs": 2,
+                    "outTooBigErrs": 0,
+                    "outNoSuchNameErrs": 0,
+                    "outBadValueErrs": 2,
+                    "outGeneralErrs": 0,
+                },
+            }
+        ],
+        "inputs": {},
+        "expected": {
+            "result": "failure",
+            "messages": ["The following SNMP error counter(s) are not found or have non-zero counter:\ninVersionErrs, inParseErrs, outBadValueErrs"],
```

Comment:
make a dict of all failure error and value as error count or not found

Reply:
Added dict in failure.